src/Sulu/Bundle/SnippetBundle/DependencyInjection/Compiler/SnippetAreaCompilerPass.php

```diff
@@ -72,23 +75,35 @@ public function process(ContainerBuilder $container)
     }
 
     /**
-     * Get area.
+     * @param array{key: string, title: array<string>} $area
+     * @param array<string> $locales
+     * @param array<string, string> $templateTitles
      *
-     * @return array
+     * @return array{key: string, template: string, title: array<string, string>}
      */
-    private function getArea($template, $area, $locales, $templateTitles)
+    private function getArea(ContainerBuilder $container, string $template, array $area, array $locales, array $templateTitles): array
     {
         $key = $area['key'];
 
         $titles = [];
+        $areaTitles = $area['title'];
 
-        foreach ($locales as $locale) {
-            $title = $templateTitles[$locale] . ' ' . \ucfirst($key);
-            if (isset($area['title'][$locale])) {
-                $title = $area['title'][$locale];
+        // If we only have one title and no locale (indexed 0) then it's a translation key
+        if (1 === \count($areaTitles) && \array_key_exists(0, $areaTitles)) {
+            $translator = $container->get('translator');
```

Comment:
@sulu/contributors normally because of how symfony recommends we avoid accessing services in compilerpasses. Does somebody have any concerns about this.

Reply:
I don't see any other way to do this. We could add a check if the container if the translator actually exists, but that's it.